packages/@aws-cdk/aws-apigateway/lib/model.ts

```diff
@@ -1,46 +1,153 @@
+
+export interface IModel extends IResource {
+  /**
+   * A name for the model. If you don't specify a name, AWS CloudFormation
+   * generates a unique physical ID and uses that ID for the model name.
+   * For more information, @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-name.html.
+   *
+   * Important
+   * If you specify a name, you cannot perform updates that require replacement
+   * of this resource. You can perform updates that require no or some interruption.
+   * If you must replace the resource, specify a new name.
+   *
+   * @attribute
+   */
+  readonly modelId: string;
 }
 
-/**
- * Represents a reference to a REST API's Empty model, which is available
- * as part of the model collection by default. This can be used for mapping
- * JSON responses from an integration to what is returned to a client,
- * where strong typing is not required. In the absence of any defined
- * model, the Empty model will be used to return the response payload
- * unmapped.
- *
- * Definition
- * {
- *   "$schema" : "http://json-schema.org/draft-04/schema#",
- *   "title" : "Empty Schema",
- *   "type" : "object"
- * }
- *
- * @see https://docs.amazonaws.cn/en_us/apigateway/latest/developerguide/models-mappings.html#models-mappings-models
- */
-export class EmptyModel implements IModel {
-    public readonly modelId = 'Empty';
+export interface ModelOptions {
+  /**
+   * The content type for the model.
+   * @default None
+   */
+  readonly contentType?: string;
+
+  /**
+   * A description that identifies this model..
+   * @default None
+   */
+  readonly description?: string;
+
+  /**
+   * A name for the model.
+   *
+   * Important
+   *  If you specify a name, you cannot perform updates that
+   *  require replacement of this resource. You can perform
+   *  updates that require no or some interruption. If you
+   *  must replace the resource, specify a new name.
+   *
+   * @default <auto> If you don't specify a name,
+   *  AWS CloudFormation generates a unique physical ID and
+   *  uses that ID for the model name. For more information,
+   *  see Name Type.
+   */
+  readonly modelName?: string;
+
+  /**
+   * The schema to use to transform data to one or more output formats.
+   * Specify null ({}) if you don't want to specify a schema.
+   */
+  readonly schema: jsonSchema.JsonSchema;
 }
 
-/**
- * Represents a reference to a REST API's Error model, which is available
- * as part of the model collection by default. This can be used for mapping
- * error JSON responses from an integration to a client, where a simple
- * generic message field is sufficient to map and return an error payload.
- *
- * Definition
- * {
- *   "$schema" : "http://json-schema.org/draft-04/schema#",
- *   "title" : "Error Schema",
- *   "type" : "object",
- *   "properties" : {
- *     "message" : { "type" : "string" }
- *   }
- * }
- */
-export class ErrorModel implements IModel {
```

Comment:
We can’t remove these classes since this will be a breaking change and we’ve stopped making breaking changes to our API. Please find a way to leave those functional, and add a @deprecated tag with instructions on what’s the correct API to use. We will remove this in the next major version.

Reply:
Done, I let them and just added them back as a static property in the `Model` class. The key here was getting the linter off on the dependency on IConstruct for the IModel property (took me some time to find the `package.json` entry.